app/ui-react/packages/auto-form/src/models.ts

```diff
@@ -21,12 +21,52 @@ export interface IFormField {
   onChange?: () => void;
 }
 
-export interface IFormProperty extends IConfigurationProperty {
+export interface IWhen {
+  value?: string;
+  id?: string;
+}
+
+export interface IFormPropertyRelation {
+  when?: IWhen[];
+  action?: string;
+}
+
+export interface IFormPropertyValue {
+  value?: string;
+  label?: string;
+}
+
+export interface IFormFieldAttributes {
+  [name: string]: any;
+}
+
+export interface IFormDefinitionProperty {
```

Comment:
I think this interface could see a good many of the props dropped? Eg. the `javaType`.
Not needed for this PR, mind, just something we could try to do eventually.

Reply:
Let me see about pulling `javaType` specifically.  Some of the other `group` or `label` maybe could be useful at some point if we want to support organizing form controls into groups, and I have to double-check `connectorValue`, `tags`, and `kind`.